pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -884,12 +893,21 @@ func podWorkloadBuilder(
 		w.WorkloadType = workloadapi.WorkloadType_POD // backwards compatibility
 		w.CanonicalName, w.CanonicalRevision = kubelabels.CanonicalService(p.Labels, w.WorkloadName)
 
+		// populate service to scope mapping
+		// TODO(jaellio): What is the ordering of Service and workload creation? Will service.Scope be populated by the time we get here?
```

Comment:
I'm pretty sure yes because the workload collection depends on the services collection